source/mir/random/engine/mersenne_twister.d

```diff
@@ -30,8 +30,8 @@ struct MersenneTwisterEngine(UIntType, size_t w, size_t n, size_t m, size_t r,
     @disable this();
     @disable this(this);
 
-    private enum UIntType upperMask = ~((cast(UIntType) 1u << (UIntType.sizeof * 8 - (w - r))) - 1);
     private enum UIntType lowerMask = (cast(UIntType) 1u << r) - 1;
+    private enum UIntType upperMask = ~lowerMask;
```

Comment:
Note here: according to https://en.wikipedia.org/wiki/Mersenne_Twister `upperMask` is supposed to be the lowest `w` bits of `~lowerMask`.